### DISCUSSION TITLE
Always run an action (no matter what state)

### DISCUSSION THREAD

**Discussion by csenio** [statelyai/xstate#1712] [Q&A] [answered]
I'm working on a project that requires complex drag and drop interactions. 

I want to record the mouse position and my current state every time a `mousedown` event happens, so I can calculate the new position of my element. 
Is it possible/logical to define a "global" mousedown action somewhere that does just that or do I have to add this mousedown event to every necessary state?

**Comment by davidkpiano** [comment, marked as answer]
There's two ways to model this. The first way is by handling the reaction to the `mousedown` event _outside_ of the machine:

```js
const service = interpret(dragDropMachine);

service.subscribe(state => {
  if (state.event.type === 'mousedown') {
    console.log('A mousedown event happened');
  }
});

service.start();
```

The second way requires just a tiny bit more thinking. Basically, you're saying this:

> When a `mousedown` event happens in state A, take this transition; in state B, take a different transition, etc.
> ALSO, when a `mousedown` event happens (regardless of those states), take this transition with this action.

The key words here are "ALSO" and "regardless" -- those signal that the "global" transition (really just an action) you want to take is a separate (or _orthogonal_) concern to the other transitions.

So, what can we use? Orthogonal (or "parallel") states!

Here is an example:

```js
const machine = createMachine({
  id: "machine",
  type: "parallel",
  states: {
    reaction: {
      on: {
        TOGGLE: {
          actions: () => {
            console.log("a toggle event happened");
          }
        }
      }
    },
    mode: {
      initial: "inactive",
      states: {
        inactive: {
          on: { TOGGLE: "active" }
        },
        active: {
          on: { TOGGLE: "inactive" }
        }
      }
    }
  }
});
```

https://codesandbox.io/s/clever-rgb-0ruc0?file=/src/index.js:254-704

**Reply by csenio** [reply]
Thanks for the answer! Parallel states is perfect for my use case!

**Reply by davidkpiano** [reply]
No problem! As a rule of thumb, remember that:

- Compound states represent at most **one transition** being "enabled"
- Parallel states can represent **zero or more transitions** being "enabled"

With state machines, everything that is done is a "transition"; even if you just want to do an action, you can consider that to be a transition (with an action) that goes nowhere (`target: undefined`).

So if you want two transitions, based on the rule of thumb above, you want parallel states.

**Comment by henninghall** [comment]
This solution solved this problem for me https://stackoverflow.com/a/65192294
